### ISSUE TITLE
Add mlflow tracking

### ISSUE THREAD

**Issue by andodet** [jrieke/traingenerator#17]
First of all thanks for the project, it's an interesting way to take a stab at reducing the amount of boilerplate needed even for fairly simple models. Secondly, it would be interesting to implement experiment/run tracking using [MLflow][1]. 

Have a working example on the `Image classification_PyTorch/` template, happy to submit a PR if you consider this of any interest.

[1]: https://www.mlflow.org/docs/latest/index.html

**Comment by jrieke** [comment]
Yeah this would be great, feel free to open a PR, then I'll have a look!
